exporter/prometheusexporter/metric_rename.go

```diff
@@ -0,0 +1,99 @@
+// Copyright The OpenTelemetry Authors
+//
+// Licensed under the Apache License, Version 2.0 (the "License");
+// you may not use this file except in compliance with the License.
+// You may obtain a copy of the License at
+//
+//       http://www.apache.org/licenses/LICENSE-2.0
+//
+// Unless required by applicable law or agreed to in writing, software
+// distributed under the License is distributed on an "AS IS" BASIS,
+// WITHOUT WARRANTIES OR CONDITIONS OF ANY KIND, either express or implied.
+// See the License for the specific language governing permissions and
+// limitations under the License.
+
+package prometheusexporter // import "github.com/open-telemetry/opentelemetry-collector-contrib/exporter/prometheusexporter"
+
+import (
+	"strings"
+
+	"go.opentelemetry.io/collector/pdata/pmetric"
+)
+
+// The map to translate OTLP units to Prometheus units
+// OTLP metrics use the c/s notation as specified at https://ucum.org/ucum.html
+// (See also https://github.com/open-telemetry/opentelemetry-specification/blob/main/specification/metrics/semantic_conventions/README.md#instrument-units)
+// Prometheus best practices for units: https://prometheus.io/docs/practices/naming/#base-units
+// OpenMetrics specification for units: https://github.com/OpenObservability/OpenMetrics/blob/main/specification/OpenMetrics.md#units-and-base-units
+var unitMap = map[string]string{
+	"s":   "seconds",
+	"Cel": "celsius",
+	"m":   "meters",
+	"By":  "bytes",
+	"V":   "volts",
+	"A":   "amperes",
+	"J":   "joules",
+	"W":   "watts",
+	"g":   "grams",
+	"Hz":  "hertz",
+	"1":   "ratio",
+}
+
+// Build a Prometheus-compliant metric name for the specified metric
+func buildPromCompliantName(metric pmetric.Metric) string {
+
+	name := strings.Map(sanitizeRune, strings.TrimSpace(metric.Name()))
+	unit := strings.TrimSpace(metric.Unit())
+	isCounter := metric.DataType() == pmetric.MetricDataTypeSum && metric.Sum().IsMonotonic()
+
+	// Remove _total suffix for now
+	// The _total suffix will be appended later for Counters
+	if isCounter {
+		name = strings.TrimSuffix(name, "_total")
+	}
+
+	if promStandardUnit, ok := unitMap[unit]; ok {
+
+		// Standard Otel unit is converted to standard Prom unit
+		promStandardUnit = "_" + promStandardUnit
+		if !strings.HasSuffix(name, promStandardUnit) {
+			name = name + promStandardUnit
+		}
+
+	} else if len(unit) == 0 || (strings.HasPrefix(unit, "{") && strings.HasSuffix(unit, "}")) {
+
+		// Do nothing with the unit
+
+	} else {
+
+		// Unit is not Prometheus-standard
+		// We will append it after best-effort sanitization
+
+		// Some special chars are converted to their probable meaning
+		promTentativeUnit := strings.ReplaceAll(unit, "/", "_per_")
```

Comment:
If we want to handle `/`, I think we need to split by `/`, map each side to a unit (using the same strategy as above), and joining the results with _per_.  So `GiBy/h` would become `disk_write_gibibyte_per_hour` (assuming we handle GiBy and h in the map).  Right now, I think we end up with `disk_write_GiBy_per_h`, which isn't great.